### ISSUE TITLE
Sliced arrays don't get updated __cuda_array_interface__ pointer

### ISSUE THREAD

**Issue by pentschev** [cupy/cupy#2128]
Consider the following example:

```python
import numpy as np
import cupy
from numba import cuda

@cuda.jit
def add(x, out):
    start = cuda.grid(1)
    stride = cuda.gridsize(1)
    for i in range(start, x.shape[0], stride):
        out[i] = x[i] + 5

x = np.arange(10)
y = cupy.arange(10)

out_x = x[2:5] + 5
out_y = np.empty_like(y[2:5])
add[1, 32](y[2:5], out_y)

print(out_x)
print(out_y)
```

this will print:

```
[7 8 9]
[5 6 7]
```

This is due to the `__cuda_array_interface__` data pointer always pointing to the base address of the memory block, therefore, Numba will always start computation from the base of memory block as well, which should have an adjusted offset.

**Comment by pentschev** [comment]
The fix in #2129 is now merged.

**Comment by leofang** [comment]
@pentschev thank you for fixing this! We also ran into this bug recently and I'm glad someone was ahead of us!
